tasks/sanity-inspect-image.yaml

```diff
@@ -0,0 +1,46 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: sanity-inspect-image
+spec:
+  description: >-
+    Get manifest data for the source image and its base image to workspace
+  params:
+    - name: SOURCE_IMAGE
+      description: the fully qualified image name
+      type: string
+  results:
+    - description: Base image the source image is built from
+      name: BASE_IMAGE
+  steps:
+  - name: inspect-image
+    image: quay.io/redhat-appstudio/hacbs-test:latest
+    script: |
+      IMAGE_MANIFEST=$(workspaces.sanity-ws.path)/image_manifest.json
+      BASE_IMAGE_MANIFEST=$(workspaces.sanity-ws.path)/base_image_manifest.json
+      IMAGE_RAW_MANIFEST=$(workspaces.sanity-ws.path)/image_raw_manifest.json
+
+      echo "Inspecting manifest for source image $(params.SOURCE_IMAGE)"
+      skopeo inspect --no-tags docker://$(params.SOURCE_IMAGE) > $IMAGE_MANIFEST
+      skopeo inspect --no-tags --raw docker://$(params.SOURCE_IMAGE) > $IMAGE_RAW_MANIFEST
+
+      echo "Getting base image manifest for source image $(params.SOURCE_IMAGE)"
+      BASE_IMAGE="$(jq -r ".annotations.\"org.opencontainers.image.base.name\"" $IMAGE_RAW_MANIFEST)"
+      if [ $BASE_IMAGE == 'null' ]; then
+        echo "Cannot get base image info from 'annotations'"
+        echo "Trying to get base image info from 'Labels'"
+        BASE_IMAGE="$(jq -r ".Labels.\"org.opencontainers.image.base.name\"" $IMAGE_MANIFEST)"
+        if [ $BASE_IMAGE == 'null' ]; then
+          echo "Cannot get base image info from 'Labels', please check the source image $(params.SOURCE_IMAGE)"
+        fi
+      fi
+      if [ -z $BASE_IMAGE ]; then
+        echo "Source image $(params.SOURCE_IMAGE) is built from scratch, so there is no base image"
+      else
+        echo "The base image is $BASE_IMAGE, get its manifest now"
+        skopeo inspect --no-tags docker://$BASE_IMAGE > $BASE_IMAGE_MANIFEST
+      fi
+      echo "$BASE_IMAGE" | tee $(results.BASE_IMAGE.path)
+  workspaces:
+  - name: sanity-ws
```

Comment:
using workspaces is very limiting, currently the pipelines does not count with `sanity-ws`, also we have access only to RWO pvc so it will not be possible to run tests in parallel. Ideally all test could be done in the task/step.

Reply:
That's very disappointing news - are we then completely limited to Tekton results to communicate between tasks?

Reply:
I am using [prototype-build-compliance](https://github.com/redhat-appstudio/build-definitions/blob/main/pipelines/base/prototype-build-compliance.yaml) pipeline as a reference. There are currently only 3 workspaces (only 1 pvc based, others are usually secrets) and tasks are usually complete.

It would be possible to re-use workspace with the source code, but in that case you cannot use it in parallel tasks.

CC @jduimovich @amfred

Reply:
@Michkov @jduimovich @amfred Would this feature alleviate the issue?

https://tekton.dev/vault/pipelines-v0.15.2/workspaces/#specifying-workspace-order-in-a-pipeline-and-affinity-assistants

Reply:
Ok. So let me just use the same solution with https://github.com/redhat-appstudio/build-definitions/pull/55 for now.